src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.ts

```diff
@@ -0,0 +1,50 @@
+import {angular, model, ng} from "entcore";
+import {ROOTS} from "../../core/const/roots";
+import {Calendar} from "../../model";
+import {IScope} from "angular";
+
+
+interface ICalendarItemProps {
+    calendar: Calendar;
+    calendarPlatforms: Array<string>;
+
+    createExternalCalendar(): (calendar: Calendar) => void;
+}
+
+interface ICalendarItemScope extends IScope {
+    vm: ICalendarItemProps;
+}
+
+class Controller implements ng.IController {
+    private onCreateExternalCalendar: () => (calendar: Calendar) => void;
+
+    constructor(private $scope: ICalendarItemScope) {
+    }
+
+    $onInit() {
+    }
+
+    $onDestroy() {
+    }
+
+    createExternalCalendar() : void {
+        this.onCreateExternalCalendar()(this.$scope.vm.calendar);
```

Comment:
```suggestion
               this.$scope.vm.createExternalCalendar()(this.$scope.vm.calendar);

```
plutôt ?

Si ça marche pas, tu pourrais l'appeler dans ton link